scheduled-jobs-alert.tf

```diff
@@ -111,3 +111,35 @@ module "delete-rejected-files-alert" {
   trigger_threshold          = 0
   resourcegroup_name         = "${azurerm_resource_group.rg.name}"
 }
+
+module "check-new-envelopes-alert" {
+  source            = "git@github.com:hmcts/cnp-module-metric-alert"
+  location          = "${azurerm_application_insights.appinsights.location}"
+  app_insights_name = "${azurerm_application_insights.appinsights.name}"
+
+  enabled    = "${var.env == "prod"}"
+  alert_name = "Check-New-Envelopes"
+  alert_desc = "Triggers when no logs from check-new-envelopes job found within timeframe."
+
+  app_insights_query =  <<EOF
+traces
+| where message startswith "Started check-new-envelopes job"
+| union (datatable(metric: real) [
+  0,
+]
+| extend timeNow = now()
+| extend day_of_week = toint(substring(tostring(dayofweek(timeNow)), 0, 1))
+| extend hour_of_the_day = datetime_part("hour", timeNow)
+| where day_of_week == 0 or day_of_week == 6 or hour_of_the_day < 9 or hour_of_the_day  > 17)
```

Comment:
what about your thoughts on below query ?

```
let startDatetime = startofday(now()); // 12AM
let startNotification = startDatetime + 9hours;
let stopNotification = startDatetime + 17hours;
let saturday = "6.00:00:00"; // Monday(0)
let sunday = "7.00:00:00";
traces
| where message startswith "Started check-new-envelopes job"
| where timestamp  >= startNotification and timestamp  <=stopNotification
| extend dayVar = dayofweek(timestamp)
| where dayVar != saturday and dayVar != sunday
| summarize count()

```

Reply:
no this query is actually not good:
- timestamp filter should be first citizen of the query
- we then check every hour for the interval between 9am 5pm. every hour (!)
- there's nothing to extend if there are no rows coming back from that query. hence another where will not work.

logically it'll always fail when it is weekend and before first envelope gets uploaded/processed during defined business hours